stl/inc/xutility

```diff
@@ -2215,19 +2213,18 @@ namespace ranges {
 
     // CONCEPT ranges::range
     template <class _Rng>
-    concept _Range_impl = requires(_Rng&& __r) {
-        _RANGES begin(static_cast<_Rng&&>(__r));
-        _RANGES end(static_cast<_Rng&&>(__r));
+    concept range = requires(_Rng& __r) {
+        _RANGES begin(__r);
+        _RANGES end(__r);
     };
 
-    template <class _Rng>
-    concept range = _Range_impl<_Rng&>;
+    // VARIABLE TEMPLATE ranges::enable_safe_range
+    template <range>
+    inline constexpr bool enable_safe_range = false;
 
-    // CONCEPT ranges::_Forwarding_range
-    // clang-format off
+    // CONCEPT ranges::safe_range
     template <class _Rng>

```

Comment:
Out of curiosity. Could that be shortened to:
```cpp
template <range _Rng>
concept safe_range = (is_lvalue_reference_v<_Rng> || enable_safe_range<remove_cvref_t<_Rng>>);
```

I guess it doesnt compose well but wouldbe interesting if the range requirement is implicit there.

Reply:
Concepts can't themselves be constrained. (http://eel.is/c++draft/temp.concept#4)

Reply:
Ah thanks for the link. I need to read up on concepts